Update script to fetch Alloy mixin dashboards

TheoBrigitte opened the pull request:
Towards: https://github.com/giantswarm/roadmap/issues/3522

This PR is an attempt at adding some automation to fetch Alloy mixin dashboards from https://github.com/grafana/alloy/tree/main/operations/alloy-mixin/dashboards

I am unsure how scripts are organized/run in this repository and would appreciate some guidance on this.

TheoBrigitte commented:
<details>
<summary>Dashboards screenshots</summary>

![image](https://github.com/user-attachments/assets/f52db9c0-48de-4f39-bcf9-fc2d13f178a8)
![image](https://github.com/user-attachments/assets/cb610ef6-9b46-4375-8607-a4fbd6f8716f)
![image](https://github.com/user-attachments/assets/e7df679c-af29-4221-934d-c16623f1cf5c)

**broken needs to be fixed**
![image](https://github.com/user-attachments/assets/f9ec6669-661c-4e5d-9b08-51f7ba764f57)

**empty since not otel receiver are configured**
![image](https://github.com/user-attachments/assets/8714c41d-4001-4643-a24d-fe14a8907081)

![image](https://github.com/user-attachments/assets/a3644320-f427-41b7-aec0-ed73f2c8e117)
![image](https://github.com/user-attachments/assets/d2e3be62-616e-49fc-bdeb-0fc69acdb876)




</details>

QuentinBisson commented:
Well I actually went into a totally different direction for mimir and loki mixins which to me is way easier to use than your script https://github.com/giantswarm/dashboards/blob/main/mimir/update.sh also because I can read it and it's not using things like grizzly which I'm not sure we need

QuentinBisson commented:
Now we can probably merge the 2 scripts together because i almost have the same script for loki. The interesting things about my script is that we can do some edits

QuentinBisson commented on scripts/update-monitoring-mixin-dashboards.sh:
```diff
@@ -11,33 +11,95 @@
+#trap "rm -rf $TMPDIR" EXIT
+echo $TMPDIR
+
+function install_tool() {
```
I feel like we use this quite a lot ( in prometheus rules for instance) is there no way to reuse the same things?

TheoBrigitte replied:
There's no easy way to reuse this AFAIK

QuentinBisson replied:
Can we not git clone a script we have somewhere else?

TheoBrigitte replied:
We could fetch something from github but I am not sure I want to got this direction.

QuentinBisson replied:
I understand but then maybe we should use a container image for this instead of downloading tools locally?

TheoBrigitte replied:
We could but do not want to add this now.

TheoBrigitte commented:
> Well I actually went into a totally different direction for mimir and loki mixins which to me is way easier to use than your script https://github.com/giantswarm/dashboards/blob/main/mimir/update.sh also because I can read it and it's not using things like grizzly which I'm not sure we need

I changed my script to something simpler and also made use of `mixtool` instead of `grizzly`, which directly produces dashboards as json files instead grizzly CRs. 

> Now we can probably merge the 2 scripts together because i almost have the same script for loki. The interesting things about my script is that we can do some edits

I've kept the script separate for now, we can revisit this later on. I wanted to keep things as "simple" as possible here.

---

I also made other modifications along the way

- Added Makefile.custom.mk to group scripts usage
  - Added `make update-mixin` to update mixin dasbhboards
  - Added `make update-mimir-mixin` to update Mimir mixin dashboards
  - Added `make update-alloy-mixin` to update Alloy mixin dashboards
  - Added `make lint-dashboards` to dashboards linting
  - Added `make install-tools` to install required tools
- Added `scripts/update-alloy-mixin.sh` to update the Alloy mixin dashboards

TheoBrigitte commented on Makefile:
```diff
@@ -23,3 +21,5 @@ include Makefile.*.mk
 help: ## Display this help.
 	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z%\\\/_0-9-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+include Makefile.*.mk
```
This is required to keep `make help` as the default command when running `make`

QuentinBisson commented on Makefile.custom.mk:
```diff
@@ -0,0 +1,25 @@
+
+dashboards = helm/dashboards/dashboards/**/*.json helm/dashboards/charts/**/*.json
+
+install-tools: ## Installat dependencies tools
```
```suggestion
install-tools: ## Install dependencies tools
```
